hack/update-bindata.sh

```diff
@@ -37,14 +36,21 @@ if ! which go-bindata &>/dev/null ; then
 fi
 
 BINDATA_OUTPUT="${KUBE_ROOT}/test/e2e/generated/bindata.go"
-go-bindata -nometadata -prefix "${KUBE_ROOT}" -o ${BINDATA_OUTPUT} -pkg generated \
+go-bindata -nometadata -prefix "${KUBE_ROOT}" -o "${BINDATA_OUTPUT}.tmp" -pkg generated \
 	-ignore .jpg -ignore .png -ignore .md \
 	"${KUBE_ROOT}/examples/..." \
 	"${KUBE_ROOT}/docs/user-guide/..." \
 	"${KUBE_ROOT}/test/e2e/testing-manifests/..." \
 	"${KUBE_ROOT}/test/images/..." \
 	"${KUBE_ROOT}/test/fixtures/..."
 
-gofmt -s -w ${BINDATA_OUTPUT}
+gofmt -s -w "${BINDATA_OUTPUT}.tmp"
 
-V=2 kube::log::info "Generated bindata file : $(wc -l ${BINDATA_OUTPUT}) lines of lovely automated artifacts"
+if ! cmp  "${BINDATA_OUTPUT}.tmp" "${BINDATA_OUTPUT}" ; then
```

Comment:
Okay cool. Glad I understand. I'm not a bash wizard, so it stood out to me. Could we get a tiny little comment with the basic explanation?

```
compare and overwrite only if different to avoid updating the timestamp and triggering make. cat redirect trick to preserve file permissions of the target file.
```

Thanks. I am done with this comment whether or not it is changed.